cmd/immuclient/immuc/setcommands.go

```diff
@@ -111,19 +100,66 @@ func (i *immuc) VerifiedSet(args []string) (string, error) {
 
 	time.Sleep(1 * time.Millisecond)
 
-	value2, err := ioutil.ReadAll(&buf)
+	vi, err := i.Execute(func(immuClient client.ImmuClient) (interface{}, error) {
+		return immuClient.VerifiedGet(ctx, key)
+	})
 	if err != nil {
 		return "", err
 	}
 
-	vi, err := i.Execute(func(immuClient client.ImmuClient) (interface{}, error) {
-		return immuClient.VerifiedGet(ctx, key)
+	return PrintKV(vi.(*schema.Entry), true, false), nil
+}
+
+func (i *immuc) Restore(args []string) (string, error) {
+	key, atRevision, err := i.parseKeyArg(args[0])
+	if err != nil {
+		return "", err
+	}
+
+	if atRevision == 0 {
+		return "please specify the key with revision to restore", nil
+	}
+
+	ctx := context.Background()
+	oldValue, err := i.Execute(func(immuClient client.ImmuClient) (interface{}, error) {
+		return immuClient.Get(ctx, key, client.AtRevision(atRevision))
+	})
+	if err != nil {
+		if strings.Contains(err.Error(), "NotFound") {
+			return fmt.Sprintf("key not found: %v ", string(key)), nil
+		}
+		rpcerrors := strings.SplitAfter(err.Error(), "=")
+		if len(rpcerrors) > 1 {
+			return rpcerrors[len(rpcerrors)-1], nil
+		}
+		return "", err
+	}
+
+	oldEntry := oldValue.(*schema.Entry)
+
+	newValue, err := i.Execute(func(immuClient client.ImmuClient) (interface{}, error) {
+		return immuClient.SetAll(ctx, &schema.SetRequest{
+			KVs: []*schema.KeyValue{{
+				Key:   oldEntry.Key,
+				Value: oldEntry.Value,
+			}},
+		})
+	})
+	if err != nil {
+		return "", err
+	}
+
+	time.Sleep(1 * time.Millisecond)
```

Comment:
pause might not be needed

Reply:
Removed in 2141bd380d07239d8c37cd721abba7a4a805df81